Clarifying language for assigning pipeline templates (#2308)
buildkite/docs: pages/pipelines/templates.md

Pipeline templates allow you to define standard pipeline step configurations to use across all the pipelines in your organization.

> When a pipeline has a template assigned, the pipeline inherits its step configuration from the template. Configuring a custom step configuration for that pipeline in the Buildkite dashboard is no longer possible. This includes any custom command steps!

Reviewer: Style: We use exclamation marks veeeery sparingly in technical writing because the excitement can feel quite inauthentic. I don't think it's needed here:

```suggestion
When a pipeline has a template assigned, the pipeline inherits its step configuration from the template. Configuring a custom step configuration for that pipeline in the Buildkite dashboard is no longer possible. This includes any custom command steps.
```

> In order to assign a pipeline template to a pipeline, you will have to enable [Requiring pipeline templates](#requiring-pipeline-templates) for your organization.

Reviewer: General tightening:

```suggestion
Before assigning a template to a pipeline, you need to [require pipeline templates](#requiring-pipeline-templates) in your organization.
```
